packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -281,16 +281,24 @@ describe('Select', () => {
         await disconnect();
     });
 
-    it('changing textContent of selected option updates display of select', async () => {
+    it('option added directly to DOM synchronously registers with Select', async () => {
         const { element, connect, disconnect } = await setup();
-        const pageObject = new SelectPageObject(element);
         await connect();
         await waitForUpdatesAsync();
-        const selectedOption = pageObject.getSelectedOption();
-        selectedOption!.textContent = 'foo';
+        const newOption = new ListOption('foo', 'foo');
+        const registerOptionSpy = spyOn(
+            element,
+            'registerOption'
+        ).and.callThrough();
+        registerOptionSpy.calls.reset();
+        element.insertBefore(newOption, element.options[0]!);
+
+        expect(registerOptionSpy.calls.count()).toBe(1);
+        expect(element.options).toContain(newOption);
+        expect(element.selectedIndex).toBe(0);
```

Comment:
From offline discussion we should make a tech debt issue to track this assertion as a known issue, that synchronous updates do not behave like native:

```
var select = Object.assign(document.createElement('select'));
var option1 = Object.assign(document.createElement('option'));
var option2 = Object.assign(document.createElement('option'), {selected: true, hidden: true, disabled: true});
select.append(option1);
select.append(option2);
console.log(select.selectedIndex); // 1, correct behavior
```